common/djangoapps/student/models.py

```diff
@@ -2750,3 +2754,22 @@ class Meta(object):
         db_table = "auth_accountrecovery"
 
     objects = AccountRecoveryManager()
+
+    @classmethod
+    def retire_user(cls, user_id):
+        """
+        # TODO: not sure this is right copied this from lms.djangoapps.verify_student.models.PhotoVerification#retire_user
+
+        Retire user as part of GDPR Phase I
+        Returns 'True' if records found
+
+        :param user_id: int
+        :return: bool
+        """
+        import pudb; pudb.set_trace()
+        # TODO: should this delete the row or make everything null
+        try:
+            cls.objects.filter(user_id=user_id).delete()
+            return True
+        except cls.DoesNotExist:
+            return False
```

Comment:
The retirement process needs to be idempotent. you may be better off just returning True here so that a re-run and not finding the record for this user wouldn't trigger a failure. It's safe to assume that if the record doesn't exist you are covered for GDPR deletion.

Reply:
Right that makes sense, Thanks!